src/corecel/random/data/DistributionData.hh

```diff
@@ -0,0 +1,131 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/random/data/DistributionData.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/OpaqueId.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Array.hh"
+#include "corecel/data/Collection.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+//! Identifier for a distribution returning a single real type
+using OnedDistributionId = OpaqueId<OnedDistributionType>;
+
+//! Identifier for a distribution returning a length three array
+using ThreedDistributionId = OpaqueId<ThreedDistributionType>;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Data for sampling a value from a delta distribution.
+ */
+template<class T>
+struct DeltaDistributionRecord
+{
+    T value{};
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Data for sampling from a normal distribution.
+ */
+struct NormalDistributionRecord
+{
+    real_type mean{0};
+    real_type stddev{1};
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Data for sampling a point uniformly on the unit sphere.
+ */
+struct IsotropicDistributionRecord
+{
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Data for sampling a point uniformly in a box.
+ */
+struct UniformBoxDistributionRecord
+{
+    using Real3 = Array<real_type, 3>;
+
+    Real3 lower{0, 0, 0};
+    Real3 upper{0, 0, 0};
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Storage for on-device sampling from arbitrary user-selected distributions.
+ */
+template<Ownership W, MemSpace M>
+struct DistributionParamsData
+{
+    //// TYPES ////
+
+    template<class T>
+    using Items = Collection<T, W, M>;
+    template<class T>
+    using OnedDistributionItems = Collection<T, W, M, OnedDistributionId>;
```

Comment:
I don't think we should be using `OnedDistributionId` for this opaque ID type, since it implies that the ID to access a `delta_real` is that ID, but it's really a separate (usually smaller) value.

Reply:
Not sure I follow... we shouldn't be accessing the types/indices with `OnedDistributionId`?